connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/network/HttpsConnection.kt

```diff
@@ -0,0 +1,47 @@
+package com.theoplayer.android.connector.uplynk.network
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runInterruptible
+import java.io.InputStreamReader
+import java.net.URL
+import javax.net.ssl.HttpsURLConnection
+
+internal const val CONNECT_TIMEOUT_IN_MS = 30000
+internal const val READ_TIMEOUT_IN_MS = 30000
+
+internal class HttpsConnection {
+    suspend fun get(urlString: String): String = runInterruptible(context = Dispatchers.IO) {
+            var result: String
+            var connection: HttpsURLConnection? = null
+            try {
+                connection = (URL(urlString).openConnection() as HttpsURLConnection)
+                    .apply {
+                        requestMethod = "GET"
+                        connectTimeout = CONNECT_TIMEOUT_IN_MS
+                        readTimeout = READ_TIMEOUT_IN_MS
+                        doInput = true
+                    }
+
+                connection.connect()
+
+                val responseCode: Int = connection.getResponseCode()
+                if (responseCode == HttpsURLConnection.HTTP_OK) {
+                    val reader = InputStreamReader(connection.inputStream)
+                    result = reader
+                        .buffered()
+                        .readText()
+                    reader.close()
```

Comment:
This part can be simplified by 
```
connection.inputStream.use {
    result = it.bufferedReader().readText()
}
```
the `.use{}` would automatically close the stream correctly whether an exception is thrown or not.

Reply:
For even more bonus points, put the assignment on the outside. 😉 
```kotlin
result = connection.inputStream.bufferedReader().use { it.readText() }
```